consensus/src/test/java/org/apache/iotdb/consensus/ratis/RatisConsensusTest.java

```diff
@@ -109,18 +118,74 @@ public DataSet read(IConsensusRequest IConsensusRequest) {
     }
 
     @Override
-    public void takeSnapshot(ByteBuffer metadata, File snapshotDir) {}
+    public void takeSnapshot(ByteBuffer metadata, File snapshotDir) {
+      String tempFilePath = snapshotDir + File.separator + ".tmp";
+      String filePath = snapshotDir + File.separator + "snapshot." + new String(metadata.array());
+
+      File tempFile = new File(tempFilePath);
+
+      try {
+        FileWriter writer = new FileWriter(tempFile);
+        writer.write(String.valueOf(integer.get()));
+        writer.close();
+        tempFile.renameTo(new File(filePath));
+      } catch (IOException e) {
+        logger.error("take snapshot failed ", e);
+      }
+    }
+
+    private Object[] getSortedPaths(File rootDir) {
+      ArrayList<Path> paths = new ArrayList<>();
+      try {
+        DirectoryStream<Path> stream = Files.newDirectoryStream(rootDir.toPath());
+        for (Path path : stream) {
+          paths.add(path);
+        }
+      } catch (IOException e) {
+        logger.error("read directory failed ", e);
+      }
+
+      Object[] pathArray = paths.toArray();
+      Arrays.sort(pathArray);
```

Comment:
When looking for the latest snapshot inside the directory, just list all filenames and sort them.